files/es/web/javascript/reference/global_objects/object/hasown/index.md

```diff
@@ -0,0 +1,162 @@
+
+```js
+const example = {};
+example.prop = 'exists';
+
+// `hasOwn` will only return true for direct properties:
+Object.hasOwn(example, 'prop');             // returns true
+Object.hasOwn(example, 'toString');         // returns false
+Object.hasOwn(example, 'hasOwnProperty');   // returns false
+
+// The `in` operator will return true for direct or inherited properties:
+'prop' in example;                          // returns true
+'toString' in example;                      // returns true
+'hasOwnProperty' in example;                // returns true
+```
+
+### Iterando sobre las propiedades de un objeto
+
+Para iterar sobre las propiedades enumerables de un objeto, _debe_ usar:
+
+```js
+const example = { foo: true, bar: true };
+for (const name of Object.keys(example)) {
+  // …
+}
+```
+
+Pero si necesita usar `for...in`, se puede usar `Object.hasOwn()` para omitir las propiedades heredadas:
+
+```js
+const example = { foo: true, bar: true };
+for (const name in example) {
+  if (Object.hasOwn(example, name)) {
+    // …
+  }
+}
+```
+
+### Comprobando si el índice de un arreglo existe
+
+Los elementos de un {{jsxref("Array")}} son definidos como propiedades directas, asi que se puede usar el método `hasOwn()` para comprobar si existe un índice en particular:
+
+```js
+const fruits = ['Apple', 'Banana', 'Watermelon', 'Orange'];
+Object.hasOwn(fruits, 3);   // true ('Orange')
+Object.hasOwn(fruits, 4);   // false - not defined
+```
+
+### Casos problematicos de hasOwnProperty
+
+Esta sección demuestra que `hasOwn()` es inmune a los problemas que afectan `hasOwnProperty`. En primer lugar, puede usarse con objetos que han reimplementado `hasOwnProperty()`:
+
+```js
+const foo = {
+  hasOwnProperty() {
+    return false;
+  },
+  bar: 'The dragons be out of office',
+};
+
+if (Object.hasOwn(foo, 'bar')) {
+  console.log(foo.bar); //true - reimplementation of hasOwnProperty() does not affect Object
+}
+```
+
+También se puede usar para probar objetos creados usando
+{{jsxref("Object.create()","Object.create(null)")}}. Estos no heredan de `Object.prototype`, por lo que `hasOwnProperty()` es inaccesible.
+
+```js
+const foo = Object.create(null);
+foo.prop = 'exists';
+if (Object.hasOwn(foo, 'prop')) {
+  console.log(foo.prop); //true - works irrespective of how the object is created.
+}
+```
+
+## Especificaciones
+
+{{Specifications}}
+
+## Compatibilidad con navegadores
+
+{{Compat}}
+
+## Véase también
+
+- [Polyfill of `Object.hasOwn` in `core-js`](https://github.com/zloirock/core-js#ecmascript-object)
+- {{jsxref("Object.prototype.hasOwnProperty()")}}
+- [Enumerability and ownership of properties](/en-US/docs/Web/JavaScript/Enumerability_and_ownership_of_properties)
+- {{jsxref("Object.getOwnPropertyNames()")}}
+- {{jsxref("Statements/for...in", "for...in")}}
+- {{jsxref("Operators/in", "in")}}
+- [JavaScript Guide: Inheritance revisited](/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain)
```

Comment:
Missing translations 

Note: we use `/es/` in links, the system automatically redirects to `en-US` if not exists

```suggestion
- [Polyfill de `Object.hasOwn` en `core-js`](https://github.com/zloirock/core-js#ecmascript-object)
- {{jsxref("Object.prototype.hasOwnProperty()")}}
- [Enumerabilidad y pertenencia de las propiedades](/es/docs/Web/JavaScript/Enumerability_and_ownership_of_properties)
- {{jsxref("Object.getOwnPropertyNames()")}}
- {{jsxref("Statements/for...in", "for...in")}}
- {{jsxref("Operators/in", "in")}}
- [Guía de JavaScript: Herencia revisada](/es/docs/Web/JavaScript/Inheritance_and_the_prototype_chain)
```